.github/workflows/validate-annotations.yml

```diff
@@ -0,0 +1,50 @@
+# this action is meant to trigger the workflow on pull requests aimed at the main branch that include changes in the namespaces/ directory
+
+name: Validate Annotations
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'namespaces/**'
+
+
+jobs:
+
+  validate-annotations:
+    # env where the job will be executed
+    name: Validate Annotations
+    runs-on: ubuntu-latest
+    steps:
+
+      #set up a Go env
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19
+      
+      # execute the validation script
+    - name: Check out code
+      uses: actions/checkout@v3
+      
+    - name: Install dependencies
+      run: |
+        go mod download
+
+    - name: Build
+      run: go build -v ./...
+
+    - name: Build Docker image
+      run: docker build -t validate-annotations .
+
+    - name: Run Validation
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: docker run --rm -e GITHUB_TOKEN=$GITHUB_TOKEN validate_annotations
+
+    - name: Run Validation (Local)
```

Comment:
We don't need this step if we are running in a docker container

Reply:
The whole steps or Run Validation (Local)?